### PR TITLE
refactor: use connection multiplexer for http and grpc server

### PR DESCRIPTION
**Pull request by sbchaos**

### CODE REVIEW

File: core/resource/service/resource_service.go [hunk 1]
```diff
@@ -157,18 +157,18 @@ func (rs ResourceService) Deploy(ctx context.Context, tnnt tenant.Tenant, store
 			r.MarkValidationFailure()
 			continue
 		}
-		r.MarkValidationSuccess()
 
 		urn, err := rs.mgr.GetURN(r)
```
Comment (sravankorumilli) [inline core/resource/service/resource_service.go:161]: provide the error information to the client as well

File: protos/odpf/optimus/core/v1beta1/job_spec.swagger.json [hunk 2]
```diff
@@ -88,6 +88,166 @@
             "in": "path",
             "required": true,
             "type": "string"
+          },
+          {
+            "name": "job.version",
```
Comment (sravankorumilli) [inline protos/odpf/optimus/core/v1beta1/job_spec.swagger.json:93]: all these changes are new which ideally shouldn't change with buf version

File: server/optimus.go [hunk 3]
```diff
@@ -190,11 +204,28 @@ func (s *OptimusServer) setupHTTPProxy() error {
...
+		grpcLn := s.mux.MatchWithWriters(
+			cmux.HTTP2MatchHeaderFieldSendSettings("content-type", "application/grpc"))
+		if err := s.grpcServer.Serve(grpcLn); err != nil {
+			s.logger.Info("shutting down grpc server")
+		}
+	}()
+
+	go func() {
+		// Start listening on http server
+		httpL := s.mux.Match(cmux.HTTP1Fast())
```
Comment (sravankorumilli) [inline server/optimus.go:217]: check if it works for http2

**Review by sravankorumilli — CHANGES_REQUESTED** [review]

File: protos/odpf/optimus/core/v1beta1/job_spec.swagger.json (hunk 2 above)
Comment (sbchaos) [reply protos/odpf/optimus/core/v1beta1/job_spec.swagger.json:93]: Should not be part of changes. Reverting changes for buf update, maybe some issue.

File: core/resource/service/resource_service.go (hunk 1 above)
Comment (sbchaos) [reply core/resource/service/resource_service.go:161]: Done

File: server/optimus.go (hunk 3 above)
Comment (sbchaos) [reply server/optimus.go]: Used check for grpc and send else to http like before.

**Review by sravankorumilli — APPROVED** [review]

**Pull request merged**